docs/input_output.md

```diff
@@ -2,85 +2,279 @@
 
 ## General Structure
 
-GIBS-in-the-Cloud uses a common format for all inputs and outputs from Tasks consisting of a JSON object which holds all necessary information about the task execution and AWS environment. Tasks return objects identical in format to their input with the exception of a task-specific `"payload"` field. Tasks may also augment their execution metadata.
+Cumulus uses a common format for all inputs and outputs from workflow steps consisting of a JSON object which holds all necessary information about the task execution and AWS environment. Tasks return objects identical in format to their input with the exception of a task-specific `"payload"` field. Tasks may also augment their execution metadata.
 
-## Data Sources
+## Input Format
+
+Below is the input format, annotated inline:
+
+    {
+      "cumulus_meta": {          // External resources accessible to the Task. Tasks should generally 
+                                 // prefer to be passed resources explicitly in their configuration rather 
+                                 // than looking up paths here. The paths being present here, however allows 
+                                 // configuration to parameterize values that are not known until the stack 
+                                 // is created.  For instance, a configuration field have the value 
+                                 // "{cumulus_meta.buckets.private}", which instructs the Task to look up the 
+                                 // private bucket while allowing the Task to remain ignorant of what buckets 
+                                 // are available.
+      
+        "stack": "<string>",     // The name of the Task's CloudFormation Task, useful as a prefix
+        "buckets": {             // Names of S3 buckets available to the app 
+          "internal": "<string>",   // The name of the bucket holding configuration and deployment data
+          "private": "<string>",    // The name of the bucket which holds internal platform data
+          "protected": "<string>",  // The name of the bucket which holds protected data
+          "public": "<string>"      // The name of the bucket which holds data to be served publicly   
```

Comment:
Suggest replacing `which holds` to `holding` in  the above 3 lines to be consistent with line 23.